### ISSUE TITLE
OSNR calculation in Transceiver class for a list of services

### ISSUE THREAD

**Issue by ojnas** [Telecominfraproject/oopt-gnpy#178]
When running `path_requests_run.py` with a service file specifying a list of services, OSNR calculation will be wrong under the following circumstances:

1. ASE or NLI is lower than 1e-20, see code in `_calc_snr` function in `class Transceiver` (this could for example happen if all elements are linear, in which case NLI = 0) and
2. OSNR has already been calculated for a different service going to the same destination transceiver.

In this case, if e.g. NLI = 0, `self.osnr_nli` and `self.snr` will not be updated. Instead, values for the previously computed path going to the same destination will be reported.

(If there was no previous path and NLI = 0, the program will instead crash which also is not ideal.)

**Comment by ggrammel** [comment]
seems not fixed. probably linked to #179 which had a regression too

changed NLI in eqpt_config.json to 0.0

ran
```
python path_requests_run.py meshTopologyToy.xls meshTopologyToy.xls
```
Result:
```
Computing path from trx R_A to trx R_C
with path constraint: ['trx R_A', 'R_B', 'trx R_C']
Computed path (roadms):['roadm R_A', 'roadm R', 'roadm R_B', 'roadm R_C']

Traceback (most recent call last):
  File "path_requests_run.py", line 144, in <module>
    line = [f'{pths[i].source} to {pths[i].destination} : ', f'{round(mean(p[-1].snr),2)}',\
  File "/anaconda3/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 2957, in mean
    out=out, **kwargs)
  File "/anaconda3/lib/python3.6/site-packages/numpy/core/_methods.py", line 82, in _mean
    ret = ret / rcount
TypeError: unsupported operand type(s) for /: 'NoneType' and 'int'
```

**Comment by ggrammel** [comment]
tried the following with and without setting gamma to 0.0 result was the same in both cases

```
>> python path_requests_run.py meshTopologyToy.xls meshTopologyToy.xls
Computing path requests meshTopologyToy.xls into JSON format
CRITICAL: missing _east_ header above other headers (hierarchical) _ ABORT
```
Then repeated the test with
```python path_requests_run.py meshTopologyExampleV2.xls meshTopologyExampleV2.xls -o tototo```
and this works. if ```meshTopologyToy.xls``` doesn't work, we could either fix or remove before committing to master

**Comment by dutc** [comment]
Can we get @EstherLerouzic to comment on these two files?

I was able to validate this as fixed with `meshTopologyExampleV2.xls`

**Comment by ggrammel** [comment]
the issue is seen with meshTopologyToy.xls, not with meshTopologyExampleV2.xls (see problem description). Pls. crosscheck with meshTopologyToy.xls

**Comment by ojnas** [comment]
@ggrammel Your issue is not related to the original issue you're commenting on. The `missing _east_ header` issue can be fixed by removing the optional `Eqpt` sheet in `meshTopologyToy.xls`. However, this topology file is probably not very relevant anymore so it might be better to just remove it (and corresponding json files) but I would leave that up to @EstherLerouzic.

@dutc It is true that the original issue was sort of fixed with with PR #180.

**Comment by EstherLerouzic** [comment]
@ojnas, following your suggestions meshTopologyToy.xls and meshTopologyToy2.xls have been removed from the examples folder in PR #215

**Comment by dutc** [comment]
Fixed in v1.2 branch. Removed:
- meshTopologyToy2.xls
- meshTopologyToy.xls
